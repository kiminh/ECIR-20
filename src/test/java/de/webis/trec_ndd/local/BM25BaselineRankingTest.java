package de.webis.trec_ndd.local;

import java.io.InputStream;

import org.approvaltests.Approvals;
import org.junit.Test;

public class BM25BaselineRankingTest {
	
	private static final String SMALL_TRAIN_TEST_SPLIT = "{\"name\": \"foo-bar\", \"train\": [], \"test\": [\"110\"]}";
	private static final String MEDIUM_TRAIN_TEST_SPLIT = "{\"name\": \"foo-bar\", \"train\": [], \"test\": [\"110\", \"abcd\", \"hello-world\"]}";
	
	@Test
	public void approveBM25RankingForSmallExampleFeatureVectorFile() {
		InputStream is = BM25BaselineRankingTest.class.getResourceAsStream("/small-example-feature-vector-file.json");
		String ranking = BM25BaselineRanking.featureVectorsToRunFile(is, SMALL_TRAIN_TEST_SPLIT);
		
		Approvals.verify(ranking);
	}
	
	@Test
	public void approveBM25RankingForMediumExampleFeatureVectorFile() {
		InputStream is = BM25BaselineRankingTest.class.getResourceAsStream("/medium-example-feature-vector-file.json");
		String ranking = BM25BaselineRanking.featureVectorsToRunFile(is, MEDIUM_TRAIN_TEST_SPLIT);
		
		Approvals.verify(ranking);
	}
	
	@Test
	public void approveBM25RankingForMediumExampleFeatureVectorFileOnSmallTrainTestSplit() {
		InputStream is = BM25BaselineRankingTest.class.getResourceAsStream("/medium-example-feature-vector-file.json");
		String ranking = BM25BaselineRanking.featureVectorsToRunFile(is, SMALL_TRAIN_TEST_SPLIT);
		
		Approvals.verify(ranking);
	}
}
