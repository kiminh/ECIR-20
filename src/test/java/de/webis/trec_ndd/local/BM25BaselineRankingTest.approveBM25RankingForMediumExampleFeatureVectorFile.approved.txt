110 Q0 id-26 1 5.040763 bm25
110 Q0 id-15 2 4.936119 bm25
110 Q0 id-10 3 4.7173357 bm25
110 Q0 id2 4 4.274645 bm25
abcd Q0 id111 1 33.33 bm25
hello-world Q0 b 1 22.22 bm25
hello-world Q0 a 2 11.11 bm25
hello-world Q0 c 3 0.33 bm25
